server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -429,14 +429,15 @@ public void test_limit_distinct_limits_outputs_to_the_group_keys_if_source_has_m
     @Test
     public void test_limit_on_join_is_rewritten_to_query_then_fetch() {
         LogicalPlan plan = plan("SELECT * FROM t1, t2 LIMIT 3");
-        assertThat(plan).isEqualTo(
-            """
-            Fetch[a, x, i, b, y, i]
-              └ Limit[3::bigint;0]
-                └ NestedLoopJoin[CROSS]
-                  ├ Collect[doc.t1 | [_fetchid] | true]
-                  └ Collect[doc.t2 | [_fetchid] | true]
-            """
+        assertThat(plan).hasOperators(
+            "Limit[3::bigint;0]",
+            "  └ NestedLoopJoin[CROSS]",
+            "    ├ Fetch[a, x, i]",
+            "    │  └ Limit[3;0]",
+            "    │    └ Collect[doc.t1 | [_fetchid] | true]",
+            "    └ Fetch[b, y, i]",
+            "      └ Limit[3;0]",
+            "        └ Collect[doc.t2 | [_fetchid] | true]"
```

Comment:
Did this break the fetch optimization here, or is this ok ? Does the fetch Operator has to be on top of the Join ? @mfussenegger @seut

Reply:
I guess we would follow up on it?

positive effect of the pushed down >>> negative effect of the `(num_node - 1) * limit` extra fetches

Reply:
Ok, this is going to be a follow-up.